kie-dmn-feel/src/main/java/org/kie/dmn/feel/runtime/decisiontables/DecisionTableImpl.java

```diff
@@ -70,42 +73,36 @@ public DecisionTableImpl(String name,
      *               decision table that are expressions derived from these parameters
      * @return
      */
-    public Object evaluate(EvaluationContext ctx, Object[] params) {
+    public FEELFnResult<Object> evaluate(EvaluationContext ctx, Object[] params) {
         if ( decisionRules.isEmpty() ) {
-            return null;
+            return FEELFnResult.ofError(new FEELEventBase(Severity.ERROR, "Decision table is empty", null));
         }
+        
+        FEEL feel = FEEL.newInstance();
+        Object[] actualInputs = resolveActualInputs( ctx, feel );
 
-        try {
-            FEEL feel = FEEL.newInstance();
-            Object[] actualInputs = resolveActualInputs( ctx, feel );
-
-            if ( ! actualInputsMatchInputValues( ctx, actualInputs ) ) {
-                return null;
-            }
+        if ( ! actualInputsMatchInputValues( ctx, actualInputs ) ) {
+            return FEELFnResult.ofError(new FEELEventBase(Severity.ERROR, "Actual inputs does not match input values", null));
```

Comment:
Need to see the effect of this error message, but as it is, it is not user friendly enough... we have to say: input X does not match valid input values [Y, Z, W]...